tools/wake-format/emitter.cpp

```diff
@@ -403,7 +426,72 @@ wcl::optional<wcl::doc> Emitter::walk_data(ctx_t ctx, CSTElement node) {
 }
 
 wcl::optional<wcl::doc> Emitter::walk_def(ctx_t ctx, CSTElement node) {
-  return walk_placeholder(ctx, node);
+  ASSERT_TOKEN(node, CST_DEF);
+  wcl::doc_builder builder;
+  CSTElement child = node.firstChildElement();
+
+  // def shouldn't follow anything other than indentation.
+  if (ctx.width > ctx.nest_level * space_per_indent) {
+    builder.append(newline(ctx));
+  }
```

Comment:
This line is suspicious, I think this should be an assert not a fix right?

Reply:
Agreed that is it weird, and I'd prefer it to be added by the parent call but not sure how to spell it out. There are plenty of `defs` that don't exceed the line width rule like `def a = b` so something like `def c = def a = b` doesn't get split. 

Either, `def` needs to check _every_ case that forces a newline after an `=` with lookahead or each case that forces a newline after an `=` needs to insert the it themselves. The solution may still be to have a `walk_rhs` function under the new API but it is unclear to me

Reply:
The only case is block I think.

Reply:
And you only have to do this in `def`, `require`, and `case`

Reply:
I came up with a reasonable solution I think. Will push it up in a second.